lib/semantics/expression.cc

```diff
@@ -2436,10 +2441,28 @@ bool ArgumentAnalyzer::IsIntrinsicRelational(RelationalOperator opr) const {
   }
 }
 
-bool ArgumentAnalyzer::IsIntrinsicNumeric() const {
-  return IsNumericTypeCategory(GetType(0)->category()) &&
-      (actuals_.size() == 1 ||
-          (AreConformable() && IsNumericTypeCategory(GetType(1)->category())));
+bool ArgumentAnalyzer::IsIntrinsicNumeric(NumericOperator opr) const {
+  std::optional<DynamicType> type0{GetType(0)};
+  if (actuals_.size() == 1) {
+    if (IsBOZLiteral(0)) {
```

Comment:
Not sure if it's significant, but the two calls to ```IsBOZLiteral(0)``` could be hoisted out of the outer ```if``` statement.